polkaj-scale/src/main/java/io/emeraldpay/polkaj/scale/writer/UInt64Writer.java

```diff
@@ -0,0 +1,31 @@
+package io.emeraldpay.polkaj.scale.writer;
+
+import io.emeraldpay.polkaj.scale.ScaleWriter;
+import io.emeraldpay.polkaj.scale.ScaleCodecWriter;
+import io.emeraldpay.polkaj.scale.reader.UInt64Reader;
+
+import java.io.IOException;
+import java.math.BigInteger;
+
+public class UInt64Writer implements ScaleWriter<BigInteger> {
+    @Override
+    public void write(ScaleCodecWriter wrt, BigInteger value) throws IOException {
+        if (value.signum() < 0) {
+            throw new IllegalArgumentException("Negative numbers are not supported by Uint64");
+        }
+        byte[] array = value.toByteArray();
+        int pos = 0;
+        // sometimes BigInteger gives an extra zero byte in the start of the array
+        if (array[0] == 0) {
+            pos++;
+        }
+        int len = array.length - pos;
+        if (len > 8) {
```

Comment:
Probably replace all 8s with `BigIntReader.UINT64_SIZE_BYTES`

Reply:
good catch, forgot to change those after testing